tests/src/gui/testqgsrubberband.cpp

```diff
@@ -128,6 +128,7 @@ void TestQgsRubberband::testBoundingRect()
   mRubberband->setIconSize( 5 ); // default, but better be explicit
   mRubberband->setWidth( 1 );    // default, but better be explicit
   mRubberband->addGeometry( geom.data(), mPolygonLayer );
+  mRubberband->setVisible( true );
```

Comment:
worth checking visibility right before and after addGeometry as that's where I think 2.6/2.8.1 used to switch visibility on (which would not work with your patch)